coders/heic.c

```diff
@@ -1319,17 +1319,49 @@ static MagickBooleanType WriteHEICImage(const ImageInfo *image_info,
             if (status == MagickFalse)
               break;
           }
-        option=GetImageOption(image_info,"heic:chroma");
-        if (option != (char *) NULL)
+      }
+#endif
+#if LIBHEIF_NUMERIC_VERSION >= 0x01080000
+    option=GetImageOption(image_info,"heic:chroma");
+    if (option != (char *) NULL)
+      {
+        error=heif_encoder_set_parameter(heif_encoder,"chroma",option);
+        status=IsHEIFSuccess(image,&error,exception);
+        if (status == MagickFalse)
+          break;
+      }
+#endif
+    options=heif_encoding_options_alloc();
+#if LIBHEIF_NUMERIC_VERSION >= 0x01100000
+    option=GetImageOption(image_info,"heic:chroma-downsampling-algorithm");
```

Comment:
Do we also want support for `preferred_chroma_upsampling_algorithm`? Also wondering if we should remove the `algorithm` word from the option.

Reply:
I'm personally indifferent about `preferred_chroma_upsampling_algorithm` because I don't expect to use it personally. I'm happy to add it though if you like.
I thought about removing the word `algorithm` too and left it in to mirror the words used by `libheif`. I'm not at all attached to keeping it though and wouldn't mind the shorter option name.